Libraries/pbxbuild/Sources/Phase/File.cpp

```diff
@@ -38,7 +38,7 @@ ResolveBuildFiles(Filesystem const *filesystem, Phase::Environment const &phaseE
             continue;
         }
 
-        std::string fileNameDisambiguator;
+        ext::optional<std::string> fileNameDisambiguator;
```

Comment:
Avoid always setting file disambiguation identifier: LGTM

I'm confused how you're getting away with changing a type here without things breaking.